v2/tools/generator/internal/codegen/pipeline/report_resource_versions.go

```diff
@@ -98,19 +98,37 @@ func (r *ResourceVersionsReport) WriteToBuffer(buffer *strings.Builder, _ string
 			lastService = svc
 		}
 
-		// For each version, write an alphabetical list of resources
-		buffer.WriteString(fmt.Sprintf("%s\n\n", pkg.PackageName()))
-
+		// For each version, write a header
+		// We use the API version of the first resource in each set, as this reflects the ARM API Version
 		resources := r.lists[pkg]
-		sort.Strings(resources)
+		sort.Slice(
+			resources,
+			func(i, j int) bool {
+				return resources[i].Name().Name() < resources[j].Name().Name()
+			})
+		firstDef := resources[0]
+		firstResource := astmodel.MustBeResourceType(firstDef.Type())
+		buffer.WriteString(
+			fmt.Sprintf(
+				"\n#### ARM version %s\n\n",
+				strings.Trim(firstResource.APIVersionEnumValue().Value, "\"")))
+
+		// write an alphabetical list of resources
 
 		for _, rsrc := range resources {
 			// Note: These links are guaranteed to work because of the Taskfile 'controller:verify-samples' target
 			path := "github.com/Azure/azure-service-operator/blob/main/v2/"
```

Comment:
Note: This will conflict with my other PR so we should just be aware.

Reply:
Thanks for the heads up.